core/system_chaincode/noop/chaincode.go

```diff
@@ -0,0 +1,105 @@
+/*
+ Copyright Digital Asset Holdings, LLC 2016 All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package noop
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric/core/chaincode/shim"
+	ld "github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/protos"
+)
+
+var logger = shim.NewLogger("noop")
+
+type ledgerHandler interface {
+	GetTransactionByUUID(txUUID string) (*protos.Transaction, error)
+}
+
+// SystemChaincode is type representing the chaincode
+// It has a ledgerHandler to be able to get TXs by ID
+type SystemChaincode struct {
+	ledgerH ledgerHandler
+}
+
+// Init initailizes the system chaincode
+func (t *SystemChaincode) Init(stub *shim.ChaincodeStub, function string, args []string) ([]byte, error) {
+	var ledger, err = ld.GetLedger()
+	if nil != err {
+		return nil, err
+	}
+	t.ledgerH = ledger
```

Comment:
Confirmed with @muralisrini that this will be an issue in system chaincodes. They behave the same as regular chaincodes in the sense that anything that is not persisted in the state should be considered stateless between invocations.

On deploy `Init` will be called and set the value of `ledgerH`. However, if the peer is stopped and then started again, the protocol says the `Init` will not be called a second time because the chaincode is already deployed. This could result in an error in the `Query` function.